guides/source/asset_pipeline.md

```diff
@@ -100,194 +76,289 @@ content into the name, usually at the end. For example a CSS file `global.css`
 
-    [Steve Souders recommends](https://www.stevesouders.com/blog/2008/08/23/revving-filenames-dont-use-querystring/),
-"...avoiding a querystring for cacheable resources". He found that in this
-case 5-20% of requests will not be cached. Query strings in particular do not
-work at all with some CDNs for cache invalidation.
+With this approach, you'll ship many small JavaScript files instead of one big JavaScript file. Thanks to HTTP/2 that no longer carries a material performance penalty during the initial transport, and in fact offers substantial benefits over the long run due to better caching dynamics.
 
-2. **The file name can change between nodes in multi-server environments.**
+How to use Import Maps as Javascript Asset Pipeline
+-----------------------------
 
-    The default query string in Rails 2.x is based on the modification time of
-the files. When assets are deployed to a cluster, there is no guarantee that the
-timestamps will be the same, resulting in different values being used depending
-on which server handles the request.
+Import Maps are the default Javascript processor, the logic of generating import maps is handled by the [`importmap-rails`](https://github.com/rails/importmap-rails) gem.
 
-3. **Too much cache invalidation**
+WARNING: Importmaps are used only for Javascript files and can not be used for CSS delivery. Check the [Sprockets section](#how-to-use-sprockets) to learn about CSS.
 
-    When static assets are deployed with each new release of code, the mtime
-(time of last modification) of _all_ these files changes, forcing all remote
-clients to fetch them again, even when the content of those assets has not changed.
+You can find detailed usage instructions on the Gem homepage, but it's important to understand the basics of `importmap-rails`.
 
-Fingerprinting fixes these problems by avoiding query strings, and by ensuring
-that filenames are consistent based on their content.
+### How it works
 
-Fingerprinting is enabled by default for both the development and production
-environments. You can enable or disable it in your configuration through the
-[`config.assets.digest`][] option.
+Importmaps are essentially a string substitution for what is referred to as "bare module specifiers". A "bare module specifier" looks like this: 
 
-More reading:
+```javascript
+import React from "react"
+```
 
-* [Optimize caching](https://developers.google.com/speed/docs/insights/LeverageBrowserCaching)
-* [Revving Filenames: don't use querystring](http://www.stevesouders.com/blog/2008/08/23/revving-filenames-dont-use-querystring/)
+This is **not compatible** with the ES Module loader spec. Instead, **to be ESM compatible**, you must provide 1 of the 3 following types of specifiers:
```

Comment:
I think for a first time user this is probably a bit technical, it's talking about concepts they may have never come across or don't care about.

For someone coming from a package.json world it's great, but I don't know if it needs to be the first thing you see here.

Reply:
I've updated this part to show import maps more like an "alias" for the library source address, as it's the simplest definition of an import map I could think of. What do you think about the newer version?